packages/shared/lib/services/connection.service.integration.test.ts

```diff
@@ -66,4 +67,103 @@ describe('Connection service integration tests', () => {
             expect(updatedMetadata).toEqual({ ...initialMetadata, ...newMetadata });
         });
     });
+
+    describe('listConnections', () => {
```

Comment:
Nice adding tests, could cover more ground by testing the API directly imo

Reply:
Yeah, I was just following the pattern of tests in the file - it probably is worth a discussion of overall testing philosophy here since there are certainly tests in the repo that follow a lot of different patterns.

Reply:
There is no philosophy tbh, those are just old tests. My opinion regarding tests: it's either very fine unit tests or e2e tests as much as we can. 
Because here you tests that it does what you want but that doesn't prevent the API from badly using the functions. Not pushing change or anything

Reply:
I guess I'm more trying to confirm that we'd rather this be an e2e test rather than a unit test (or integration since it hits the DB, I'm not sure how we're thinking about that name).